wasm/src/lib.rs

```diff
@@ -179,14 +164,12 @@ pub unsafe fn _decode_from_raw<T: Decode>(ptr: WasmUsize, len: WasmUsize) -> T {
 ///
 /// It's safe to call this function as long as it's safe to construct, from the given
 /// pointer, `Box<[u8]>` containing the encoded object
-// `WasmUsize` is always pointer sized
-#[allow(clippy::cast_possible_truncation)]
 unsafe fn _decode_from_raw_in_range<T: Decode>(
-    ptr: WasmUsize,
-    len: WasmUsize,
+    ptr: *const u8,
+    len: usize,
     range: RangeFrom<usize>,
 ) -> T {
-    let bytes = Box::from_raw(core::slice::from_raw_parts_mut(ptr as *mut _, len as usize));
+    let bytes = Box::from_raw(core::slice::from_raw_parts_mut(ptr as *mut _, len));
```

Comment:
this is potentially UB, though I think it's not

Reply:
While I'd err on the side of caution, I think you're right in it not being intrinsically UB.

Reply:
If you really want to be pedantic, maybe add an `assert`, that checks for conversion errors.